testutils/thriftarg2test/arg2_kv.go

```diff
@@ -21,3 +21,16 @@ func BuildKVBuffer(kv map[string]string) []byte {
 	}
 	return buf[:wb.BytesWritten()]
 }
+
+// ReadKVBuffer converts an arg2 buffer to a string map
+func ReadKVBuffer(b []byte) map[string]string {
+	rbuf := typed.NewReadBuffer(b)
+	nh := rbuf.ReadUint16()
+	retMap := make(map[string]string, nh)
+	for i := uint16(0); i < nh; i++ {
+		key := rbuf.ReadLen16String()
+		val := rbuf.ReadLen16String()
+		retMap[key] = val
+	}
+	return retMap
```

Comment:
should we verify if there's any data left in the `[]byte`? we should verify that the bytes are fully consumed for testing